crates/re_viewport/src/space_view.rs

```diff
@@ -98,23 +99,42 @@ impl SpaceViewBlueprint {
     }
 
     pub fn try_from_db(path: &EntityPath, blueprint_db: &StoreDb) -> Option<Self> {
-        let SpaceViewComponent {
+        re_tracing::profile_function!();
+
+        let query = LatestAtQuery::latest(Timeline::default());
+
+        let re_types::blueprint::archetypes::SpaceViewBlueprint {
             display_name,
             class_identifier,
             space_origin,
             entities_determined_by_user,
             contents,
-        } = blueprint_db
-            .store()
-            .query_timeless_component::<re_types::blueprint::components::SpaceViewComponent>(path)
-            .map(|c| c.value)?
-            .0;
+        } = query_archetype(blueprint_db.store(), &query, path)
+            .and_then(|arch| arch.to_archetype())
+            .map_err(|err| {
+                if cfg!(debug_assertions) {
+                    re_log::error!("Failed to load SpaceView blueprint: {err}.");
+                } else {
+                    re_log::debug!("Failed to load SpaceView blueprint: {err}.");
+                }
+                err
+            })
+            .ok()?;
 
         let id = SpaceViewId::from_entity_path(path);
 
-        let class_identifier = class_identifier.as_str().into();
+        let space_origin = space_origin.map_or_else(EntityPath::root, |origin| origin.0.into());
+
+        let class_identifier = class_identifier.0.as_str().into();
+
+        let display_name = display_name.map_or_else(
+            || format!("{class_identifier} ({space_origin})"),
+            |v| v.0.to_string(),
+        );
```

Comment:
no strong opinion but I think that's a very different default than what we had before.
I'd say at least only take the last path fragment of `space_origin`, otherwise the default display name gets very lengthy